lib/jvm.sh

```diff
@@ -37,11 +37,24 @@ get_jdk_version() {
 }
 
 get_full_jdk_version() {
-  local jdkVersion="${1:?}"
+  # The version argument can potentially have a prefix which denotes the
+  # OpenJDK distribution. This function only normalizes the actual version
+  # and keeps the prefix intact.
+  IFS='-' read -r prefix version <<<"${1:?}"
+
+  if [ -z "${version}" ]; then
+    # If the version variable is empty, there is no prefix and the
+    # version was stored in the prefix variable.
+    version="${prefix}"
+  else
+    # When there is a prefix, emit it before emitting the normalized
+    # version to keep it untouched by this function.
+    echo -n "${prefix}-"
+  fi
 
-  case "${jdkVersion}" in
+  case "${version}" in
   "7" | "1.7") echo "${DEFAULT_JDK_1_7_VERSION}" ;;
-  "8" | "1.8") echo "{$DEFAULT_JDK_1_8_VERSION}" ;;
```

Comment:
Ah I missed this when reviewing #237. It seems this has no test coverage? Should it?

Reply:
It actually has test coverage in `java_spec.rb` where we install with both `8` and `1.8` as the version strings - and it works, even though this line would resolve to smth. along the lines of `{1.8.0_332}`. I haven't investigated yet why it works, that will be a separate work stream.